packages/lab/src/stepper-input/StepperInput.tsx

```diff
@@ -1,47 +1,77 @@
 import { clsx } from "clsx";
-import React, { forwardRef, ReactNode, useRef } from "react";
-import { Button, ButtonProps, makePrefixer } from "@salt-ds/core";
+import { FocusEventHandler, forwardRef, useRef } from "react";
+import { Button, makePrefixer, Input, InputProps } from "@salt-ds/core";
 import { useComponentCssInjection } from "@salt-ds/styles";
 import { useWindow } from "@salt-ds/window";
-import { RefreshIcon, TriangleDownIcon, TriangleUpIcon } from "@salt-ds/icons";
-import {
-  InputLegacy as Input,
-  InputLegacyProps as InputProps,
-} from "../input-legacy";
-import { useActivationIndicatorPosition } from "./internal/useActivationIndicatorPosition";
+import { TriangleDownIcon, TriangleUpIcon } from "@salt-ds/icons";
 import { useStepperInput } from "./useStepperInput";
 
 import stepperInputCss from "./StepperInput.css";
 
 const withBaseName = makePrefixer("saltStepperInput");
 
-export interface StepperInputProps {
-  ButtonProps?: Partial<ButtonProps>;
-  InputProps?: Partial<InputProps>;
+export interface StepperInputProps
+  extends Omit<InputProps, "onChange" | "readOnly" | "emptyReadOnlyMarker"> {
+  /**
+   * A multiplier applied to the `step` when the value is incremented or decremented using the PageDown/PageUp keys.
+   */
   block?: number;
-  className?: string;
+  /**
+   * The number of decimal places to display.
+   */
   decimalPlaces?: number;
+  /**
+   * Sets the initial default value of the component.
+   */
   defaultValue?: number;
-  liveValue?: number;
+  /**
+   * The maximum value that can be selected.
+   */
   max?: number;
+  /**
+   * The minimum value that can be selected.
+   */
   min?: number;
-  onBlur?: (event: React.FocusEvent<HTMLDivElement>) => void;
-  onFocus?: (event: React.FocusEvent<HTMLDivElement>) => void;
+  /**
+   * Whether to hide the stepper buttons. Defaults to `false`.
+   */
+  hideButtons?: boolean;
+  /**
+   * Callback when stepper input loses focus.
+   */
+  onBlur?: FocusEventHandler<HTMLInputElement>;
+  /**
+   * Callback when stepper input value is changed.
+   */
   onChange?: (changedValue: number | string) => void;
```

Comment:
Any possibility to expose native `onChange` event as well?

Reply:
I don't think there is always a native input onchange event as the value can be changed with the buttons

Reply:
What's the case `string` will be returned instead of a number, if we're parsing using regex?

Reply:
you need to be able type "1." but if it's always a number that isn't valid